app.js

```diff
@@ -1,19 +1,318 @@
+  const cart = carts.find(cart => cart.userId === userId);
+
+  if (!cart) {
+    const userProducts = [];
+    userProducts.push(product);
+    const cart = {
+      cartId: randomUUID(),
+      userId,
+      userProducts
+    }
+    carts.push(cart)
+    res.status(200).send(cart);
+  } else {
+    cart.userProducts.push(product);
+    res.status(200).send(carts);
+  }
+
+})
+
+app.delete('/cart/:id', isAuth, (req, res) => {
+  const userId = res.locals;
+  const cart = carts.find(c => c.userId === userId)
+  if (!cart) {
+    throw new CustomError(404, "Cart not found!");
+  }
+
+  const productIndex = cart.userProducts.findIndex(p => p.id == req.params.id);
+  if (productIndex === -1) {
+    throw new CustomError(404, "Product not found in cart");
+  }
+
+  cart.userProducts.splice(productIndex, 1);
+  res.status(200).send(carts);
+
+})
+
+app.post('/cart/checkout', isAuth, (req, res) => {
+  const userId = res.locals;
+  const cart = carts.find(c => c.userId === userId);
+  if (!cart) {
+    throw new CustomError(404, "Cart not found")
+  }
+
+  const order = {};
+  const orderId = randomUUID();
+  const totalPrice = cart.userProducts.reduce((acc, curr) => acc + curr.price, 0);
+
+  if (order) {
+    order.id = orderId;
+    order.userId = userId;
+    order.products = cart.userProducts;
+    order.price = totalPrice;
+
+    orders.push(order);
+    res.status(200).send(orders)
+  }
+
+})
+
+// new product
+
+app.post('/product', (req, res) => {
+  const id = randomUUID();
+  const newProduct = {
+    id: id,
+    name: req.body.name,
+    description: req.body.description,
+    price: req.body.price,
+    images: [],
+    video: []
+  };
+  const data = JSON.stringify(newProduct);
+  fs.writeFile('products.store.json', data, (err) => {
+    if (err) {
+      throw new CustomError(401, "Product not found!")
+    }
+  })
+  res.status(200).send(newProduct);
+})
+
+//
+
+app.post('/product/:id/image/upload', (req, res) => {
+
+  const imageName = randomUUID();
+  const filePath = path.join(__dirname, 'images', `${imageName}.jpg`)
+  const imageStream = fs.createWriteStream(filePath);
+
+  imageStream.on('fileUploadStart', () => {
+    events.emit('File upload has started')
+    logToFile('File upload has started');
+  })
+  req.pipe(imageStream);
+
+  imageStream.on('fileUploadEnd', () => {
+    events.emit('File has been uploaded')
+    logToFile('File has been uploaded');
+  })
+
+  imageStream.on('fileUploadFailed', () => {
+    events.emit('File write error')
+    logToFile('File write error');
+  })
+
+  fs.readFile('./products.store.json', 'utf8', (err, data) => {
+    if (err) {
+      throw new CustomError(401, "Ошибка чтения json файла");
+    }
+
+    const product = JSON.parse(data);
+    product.images.push(`${imageName}.jpg`);
+    const productData = JSON.stringify(product);
+    fs.writeFile('products.store.json', productData, (err) => {
+      if (err) {
+        throw new CustomError(401, "Ошибка записи файла");
```

Comment:
401 code is not suitable here because it tells that user is unauthorized, message is strange too